extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -10,46 +10,79 @@ class NodeImage extends Node {
 	const CAPTION_TAG_NAME = 'caption';
 	const MEDIA_TYPE_VIDEO = 'VIDEO';
 
+	private function getImage($key, $caption, $alt) {
+		$title = $this->getImageAsTitleObject( $key );
+		$file = $this->getFilefromTitle( $title );
+		if ( $title instanceof \Title ) {
+			$this->getExternalParser()->addImage( $title->getDBkey() );
+		}
+		$ref = null;
+		wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
+		$item = [
+			'url' => $this->resolveImageUrl( $file ),
+			'name' => ( $title ) ? $title->getText() : '',
+			'key' => ( $title ) ? $title->getDBKey() : '',
+			'alt' => $alt,
+			'caption' => $caption,
+			'ref' => $ref
+		];
+		if ( $this->isVideo( $file ) ) {
+			$item = $this->videoDataDecorator( $item, $file );
+		}
+		return $item;
+	}
+
 	public function getData() {
 		if ( !isset( $this->data ) ) {
 			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
-			}
-
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
-			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption,
-				'ref' => $ref
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+			$imageDataLower = strtolower($imageData);
+
+			if ( strpos( $imageDataLower, '-tabber-' ) !== false || strpos( $imageDataLower, '-gallery-' ) !== false ) {
```

Comment:
Parsing HTML output carries a lot of issues with it, for example gallery produces different output depending on params used (slider, slideshow, etc.). However by using PortableInfoboxDataBag we can get wikitext, additionally we could refactor galleries to extract parsing logic and reuse it. That way we are immune to gallery parsing changes and we don't need to make so many regexes and additional parsing. The same could be applied to tabber extension. For inside template usage it could go the same flow, as raw wikitext is provided in template. What do you think about that?